components/support/sql/src/open_database.rs

```diff
@@ -35,89 +42,106 @@ pub enum Error {
 
 pub type Result<T> = std::result::Result<T, Error>;
 
-pub trait MigrationLogic {
+pub trait ConnectionInitializer {
     // Name to display in the logs
     const NAME: &'static str;
 
     // The version that the last upgrade function upgrades to.
     const END_VERSION: u32;
 
-    // Runs first, outside of a transaction
-    fn setup_pragmas(&self, _conn: &Connection) -> Result<()> {
-        Ok(())
-    }
+    // Functions called only for writable connections all take a Transaction
+    // Initialize a newly created database to END_VERSION
+    fn init(&self, tx: &Transaction<'_>) -> Result<()>;
```

Comment:
+1 to explicitly stating which methods are inside a transaction vs a connection.